### PR TITLE
improvement(linter/oxc): using facades for barrel detections.

### LINKED ISSUE #3069: feat(linter): stabilize no-barrel-file
Need to test in real projects.

Currently seeing false positives, and also panics.

### PR DESCRIPTION
**Pull request by rzvxa**
closes #3069 

It is similar to [this](https://gist.github.com/developit/a306951af9c0cfdf5925f126428887eb#file-no-barrel-js-L126), I've dialed back some package resolving to make it more general, We might want to invest in creating a package resolver(let me know if we already have the infrastructure for it and I've missed it).

### CODE REVIEW

**Comment by rzvxa** [comment]
* **#3215** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/3215?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @rzvxa and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: crates/oxc_linter/src/rules/oxc/no_barrel_file.rs [hunk 1]
```diff
@@ -60,23 +78,90 @@ impl Rule for NoBarrelFile {
...
+                NoBarrelFileDiagnostic::BarrelFile(program.span)
+            };
+
+            ctx.diagnostic(diag);
         }
     }
+
+    fn run(&self, node: &AstNode<'_>, ctx: &LintContext<'_>) {
+        let AstKind::ImportDeclaration(import) = node.kind() else { return };
+        if is_facade_import(ctx.file_path().to_str().unwrap(), import.source.value.as_str()) {
```
Comment (Boshen) [inline crates/oxc_linter/src/rules/oxc/no_barrel_file.rs:99]: I can simply this, leave this to me.

**Comment by Boshen** [comment]
Now I understand what this is doing once it's written in Rust 😅

File: crates/oxc_linter/src/rules/oxc/no_barrel_file.rs (hunk 1 above)
Comment (rzvxa) [reply crates/oxc_linter/src/rules/oxc/no_barrel_file.rs]: Feel free to take over this PR, I'll start working on other stuff. Let me know if there is anything you want me to change.

**Pull request closed without merging**